File: in_Java/0102 Binary Tree Level Order Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        if(root == null) return res;
        q.add(root);
        List<Integer> cur;
        TreeNode x;
        int n;
        while(q.size() != 0){
            cur = new ArrayList<>();
            n = q.size();
            for(int i = 0; i < n; i++){
                x = q.poll();
                if(x.left != null) q.add(x.left);
                if(x.right != null) q.add(x.right);
                cur.add(x.val);
            }
            res.add(cur);
        }
        return res;
    }
}

File: in_Python/0102 Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        q = []
        res = []
        if root == None:
            return res
        q.append(root)
        while len(q) != 0:
            n = len(q)
            cur = []
            for i in range(n):
                head = q.pop(0)
                if head.left != None:
                    q.append(head.left)
                if head.right != None:
                    q.append(head.right)
                cur.append(head.val)
            res.append(cur)
        return res
        